### PR TITLE
Fix app insights logging

### PR DESCRIPTION
**Pull request by adusumillipraveen**
- App insights wasn't working as secret was not mounted to application. 
- Fixed couple of other things while there.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -161,9 +161,12 @@ dependencies {
     implementation group: 'uk.gov.hmcts.reform', name: 'health-spring-boot-starter', version:'0.0.5'
     implementation group: 'uk.gov.hmcts.reform', name: 'logging', version: versions.reformLogging
     implementation group: 'uk.gov.hmcts.reform', name: 'logging-appinsights', version: versions.reformLogging
-    implementation group: 'uk.gov.hmcts.reform', name: 'logging-httpcomponents', version: versions.reformLogging
     implementation group: 'uk.gov.hmcts.reform', name: 'logging-spring', version: versions.reformLogging
     implementation group: 'uk.gov.hmcts.reform', name: 'service-auth-provider-client', version: versions.serviceTokenGenerator
+    implementation group: 'uk.gov.hmcts.reform', name: 'properties-volume-spring-boot-starter', version: '0.1.0'
+
+    implementation group: 'org.apache.logging.log4j', name: 'log4j-api', version: '2.13.3'
```
Comment (adusumillipraveen) [inline build.gradle:168]: Just aligning to spring-boot-template

File: charts/finrem-cos/values.yaml [hunk 2]
```diff
@@ -13,8 +13,4 @@ java:
   keyVaults:
     finrem:
       secrets:
-        - finrem-doc-s2s-auth-secret
```
Comment (adusumillipraveen) [inline charts/finrem-cos/values.yaml:16]: These weren't used  as app works without them ( even in production)

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -42,7 +42,6 @@ management.health.diskspace.enabled=true
 # Enable the metrics endpoint
 management.endpoint.metrics.enabled=true
 
-azure.app_insights_key=${AZURE_APPLICATIONINSIGHTS_INSTRUMENTATIONKEY:false}
```
Comment (adusumillipraveen) [inline src/main/resources/application.properties:45]: This looks like wrong key anyway, but better to not have a dummy default in this case.

**Review by johnoneill46 — APPROVED** [review]

**Review by HarryH96 — APPROVED** [review]

File: charts/finrem-cos/values.yaml [hunk 4]
```diff
@@ -13,8 +13,4 @@ java:
...
-        - idam-secret
-        - username-solicitor
```
Comment (shaunganley) [inline charts/finrem-cos/values.yaml:18]: @johnoneill46 from quick look, functional tests won't pass without these:
```
@Value("${idam.username}")
    private String idamUserName;
```

Comment (adusumillipraveen) [reply charts/finrem-cos/values.yaml:18]: Functional tests don't need these to be defined in application. They are defined in Jenkinsfile_CNP. see https://github.com/hmcts/cnp-jenkins-library#secrets-for-functional--smoke-testing

**Pull request merged**